helix-view/src/view.rs

```diff
@@ -165,6 +166,67 @@ impl View {
         Some(Position::new(row, col))
     }
 
+    /// Verifies whether a screen position is inside the view
+    /// Returns true when position is inside the view
+    pub fn verify_screen_coords(&self, row: usize, column: usize) -> bool {
+        // 2 for status
+        if row < self.area.y as usize || row > self.area.y as usize + self.area.height as usize - 2
+        {
+            return false;
+        }
+
+        // TODO: not ideal
+        const OFFSET: usize = 7; // 1 diagnostic + 5 linenr + 1 gutter
+
+        if column < self.area.x as usize + OFFSET
+            || column > self.area.x as usize + self.area.width as usize
+        {
+            return false;
+        }
+        true
+    }
+
+    /// Translates a screen position to position in the text document.
+    /// Returns a usize typed position in bounds of the text if found in this view, None if out of view.
+    pub fn pos_at_screen_coords(&self, doc: &Document, row: usize, column: usize) -> Option<usize> {
+        if !self.verify_screen_coords(row, column) {
+            return None;
+        }
+
+        let text = doc.text();
+        let line_number = row - self.area.y as usize + self.first_line;
+
+        if line_number > text.len_lines() - 1 {
+            return Some(text.len_chars() - 1);
```

Comment:
Sorry for the run-around on this: now that #376 has been merged, this should just be `Some(text.len_chars())`, because to-the-right of the last character is now the last valid document position.

Reply:
Thanks, I did not repeat test with short file after branch update.